main.tf

```diff
@@ -146,7 +146,7 @@ resource "aws_efs_mount_target" "this" {
 
   file_system_id  = aws_efs_file_system.this[0].id
   ip_address      = try(each.value.ip_address, null)
-  security_groups = var.create_security_group ? concat([aws_security_group.this[0].id], try(each.value.security_groups, [])) : try(each.value.security_groups, null)
+  security_groups = var.create_security_group ? concat([aws_security_group.this[0].id], try(each.value.security_groups, [])) : try(each.value.security_groups, var.security_groups)
```

Comment:
we already support this via the `each.value.security_groups` shown above

so in your mount targets:

```hcl
  # Truncated for brevity
  mount_targets = {
    one = {
      security_groups = [ #add them here]
    }
  }